src/runtime/texture_pool.cc

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
```

Comment:
consider move to src/runtime/opencl for now(as texture utils is only needed by OpenCL atm and we can consider add vulkan support later). 

With vulkan support we can move to src/runtime/texture or make it as a header util that both OpenCL and vk can include